### PR TITLE
Add functionality to send/recv matrices

### PR DESCRIPTION
**Pull request by jeanlucf22**
MPI point-to-point send/recv for BML matrices
of all formats

---
This change is [Reviewable](https://reviewable.io/reviews/lanl/bml/438)

### CODE REVIEW

**Comment by jeanlucf22** [comment]
@nicolasbock Looks like we still have an issue with finding a MPI library...

**Review by nicolasbock — APPROVED** [review]
Thanks @jeanlucf22 !

I think the only thing missing is to test for the new MPI tests in CI. I have pushed a small change that enables those tests.

Thanks!

File: tests/C-tests/CMakeLists.txt [hunk 1]
```diff
@@ -109,41 +108,49 @@ function(test_formats ${formats})
...
-      allocate)
-else()
-  set(testlist
-    add
-    adjacency
-    adjungate_triangle
+function(test_formats_mpi ${formats})
+  foreach(T ${formats} )
+    foreach(P single_real double_real single_complex double_complex)
+      add_test(MPI-C-${N}-${T}-${P} mpirun -np 4 ./bml-test -n ${N} -t ${T} -p ${P})
```
Comment (nicolasbock) [inline tests/C-tests/CMakeLists.txt:114]: For some reason, `mpirun` is very sensitive to the path in the executable. When you add `./bml-test` with the relative path it starts working.

**Comment by jeanlucf22** [comment]
@nicolasbock Thanks for merging my PR. But to me it still looks like it did not pass the CI tests...

**Comment by nicolasbock** [comment]
Hi @jeanlucf22 ! Where do you see a failure? I looked at

https://github.com/lanl/bml/runs/1751563839?check_suite_focus=true#step:4:1364

The latest CI tests and it looks like all MPI tests have passed. Did you find other tests that are failing?

Thanks!

**Comment by jeanlucf22** [comment]
I was looking at
https://app.circleci.com/pipelines/github/lanl/bml/506/workflows/9faaf37f-dcaa-44ce-875f-5c4e2e451a8e/jobs/5963
I got there by clicking on the red cross showing failure in the commit list at
https://github.com/lanl/bml/commits/master

**Comment by nicolasbock** [comment]
The container we are using for the CircleCI tests are not up to date. Ah. I am trying to remove CircleCI from this project (#437) so we don't have to maintain two CI systems.

**Pull request merged**